### PR TITLE
Push after running `neuron_parallel_compile`

### PR DESCRIPTION
**Pull request by michaelbenayoun**
This PR adds support for pushing to the cache repository on the Hugging Face Hub after using `neuron_paralell_compile`. 

Basically, when running a training after doing `neuron_parallel_compile`, all the compilation files will be pushed to the Hub before starting actual training.

cc @5cp

### CODE REVIEW

File: optimum/neuron/utils/cache_utils.py [hunk 1]
```diff
@@ -41,20 +41,19 @@
...
 
 
-if TYPE_CHECKING:
-    from transformers import PretrainedConfig, PreTrainedModel
-
-
 logger = logging.get_logger()
 
+# SHOULD_LOG = should_log()
+
```
Comment (dacorvo) [inline optimum/neuron/utils/cache_utils.py:56]:
```suggestion
```

File: optimum/neuron/utils/misc.py [hunk 2]
```diff
@@ -38,12 +38,21 @@
...
 
 from ...utils import logging
+from .import_utils import is_torch_xla_available
 from .require_utils import requires_safetensors
 
 
 logger = logging.get_logger()
 
 
+def should_log() -> bool:
```
Comment (dacorvo) [inline optimum/neuron/utils/misc.py:48]: This is very specific to neuronx_distributed, and not really related to logging. Can you move it under the neuronx_distributed directory, and choose a name that clearly states that this verifies we are in the process of the first worker (that's what I understood).

File: optimum/neuron/utils/misc.py [hunk 3]
```diff
@@ -45,7 +45,7 @@
 logger = logging.get_logger()
 
 
-def should_log() -> bool:
+def should_current_worker_log() -> bool:
```
Comment (dacorvo) [inline optimum/neuron/utils/misc.py:48]: It still has nothing to do with log. I would call it is_main_worker, wdyt ?

**Review by dacorvo — APPROVED** [review]
Maybe change the name, otherwise LGTM ...

File: optimum/neuron/utils/misc.py (hunk 3 above)
Comment (michaelbenayoun) [reply optimum/neuron/utils/misc.py:48]: Alright, doing it.

**Pull request merged**